app/src/main/kotlin/at/bitfire/davdroid/sync/TaskSyncer.kt

```diff
@@ -11,109 +11,92 @@ import android.content.SyncResult
-            val updateColors = accountSettings.getManageCalendarColors()
-            val newCollections = HashMap(remoteCollections)
-            for (list in DmfsTaskList.find(account, taskProvider, LocalTaskList.Factory, null, null))
-                list.syncId?.let {
-                    val url = it.toHttpUrl()
-                    val info = remoteCollections[url]
-                    if (info == null) {
-                        logger.fine("Deleting obsolete local task list $url")
-                        list.delete()
-                    } else {
-                        // remote CollectionInfo found for this local collection, update data
-                        logger.log(Level.FINE, "Updating local task list $url", info)
-                        list.update(info, updateColors)
-                        // we already have a local task list for this remote collection, don't create a new local task list
-                        newCollections -= url
-                    }
-                }
+    private lateinit var taskProvider: TaskProvider
 
-            // 3. create new local task lists
-            for ((_,info) in newCollections) {
-                logger.log(Level.INFO, "Adding local task list", info)
-                LocalTaskList.create(account, taskProvider, info)
-            }
+    override val serviceType: String
+        get() = Service.TYPE_CALDAV
+    override fun localCollections(provider: ContentProviderClient): List<LocalTaskList>
+        = DmfsTaskList.find(account, taskProvider, LocalTaskList.Factory, null, null)
+    override fun localSyncCollections(provider: ContentProviderClient): List<LocalTaskList>
+        = DmfsTaskList.find(account, taskProvider, LocalTaskList.Factory, "${TaskContract.TaskLists.SYNC_ENABLED}!=0", null)
 
-            // 4. sync local task lists
-            val localTaskLists = DmfsTaskList
-                .find(account, taskProvider, LocalTaskList.Factory, "${TaskContract.TaskLists.SYNC_ENABLED}!=0", null)
-            for (localTaskList in localTaskLists) {
-                logger.info("Synchronizing task list #${localTaskList.id} [${localTaskList.syncId}]")
+    override fun prepare(provider: ContentProviderClient): Boolean {
 
-                val url = localTaskList.syncId?.toHttpUrl()
-                remoteCollections[url]?.let { collection ->
-                    val syncManager = tasksSyncManagerFactory.tasksSyncManager(
-                        account,
-                        accountSettings,
-                        httpClient.value,
-                        extras,
-                        authority,
-                        syncResult,
-                        localTaskList,
-                        collection
-                    )
-                    syncManager.performSync()
-                }
-            }
+        // Acquire task provider
+        val providerName = TaskProvider.ProviderName.fromAuthority(authority)
+        taskProvider = try {
```

Comment:
… and make this a local `val`, shouldn't it work?

Reply:
If we would not need the taskProvider then we would not have to save it here either (not at all in fact).